deploy/eck-integrations/templates/eck-agent/elastic-agent-daemonset.yaml

```diff
@@ -0,0 +1,95 @@
+---
+apiVersion: agent.k8s.elastic.co/v1alpha1
+kind: Agent
+metadata:
+  name: eck-agent-daemonset
+  namespace: {{ .namespace | default $.Release.Namespace | quote }}
+  labels:
+    {{- include "elasticagent.labels" $ | nindent 4 }}
+  annotations:
+    {{- with .Values.eck_agent.annotations }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  version: {{ required "An Elastic Agent version is required" .Values.eck_agent.version }}
+  configRef:
+    secretName: eck-agent-daemonset
+  daemonSet:
+    podTemplate:
+      metadata:
+        annotations:
+          checksum/config: {{ include (print $.Template.BasePath "/eck-agent/elastic-agent-daemonset-secret.yaml") . | sha256sum }}
+      spec:
+        automountServiceAccountToken: true
+        serviceAccountName: eck-agent
+        containers:
+        - name: agent
+          securityContext:
+            runAsUser: 0
+            {{- if eq .Values.cloudDefend.enabled true}}
+            capabilities:
+              add:
+                - BPF # (since Linux 5.8) allows loading of BPF programs, create most map types, load BTF, iterate programs and maps.
+                - PERFMON # (since Linux 5.8) allows attaching of BPF programs used for performance metrics and observability operations.
+                - SYS_RESOURCE # Allow use of special resources or raising of resource limits. Used by 'Defend for Containers' to modify 'rlimit_memlock'
+            {{- end}}
+          resources:
+            limits:
+              memory: 700Mi
+            requests:
+              cpu: 100m
+              memory: 400Mi
+          env:
+          - name: NODE_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: spec.nodeName
+        {{- if eq .Values.cloudDefend.enabled true}}
+        volumeMounts:
+        - name: proc
+          mountPath: /hostfs/proc
+          readOnly: true
+        - name: cgroup
+          mountPath: /hostfs/sys/fs/cgroup
+          readOnly: true
+        - name: varlibdockercontainers
+          mountPath: /var/lib/docker/containers
+          readOnly: true
+        - name: varlog
+          mountPath: /var/log
+          readOnly: true
+        - name: etc-full
+          mountPath: /hostfs/etc
+          readOnly: true
+        - name: var-lib
+          mountPath: /hostfs/var/lib
+          readOnly: true
+        - name: sys-kernel-debug
+          mountPath: /sys/kernel/debug
+        - name: elastic-agent-state
+          mountPath: /usr/share/elastic-agent/state
+        {{- end}}
+        {{- if eq .Values.cloudDefend.enabled true}}
+        volumes:
```

Comment:
@pkoutsovasilis I think we should use a different hostPath for this chart than is used in the other Agent manifests. That way if a user upgrades from managed to standalone agent (in this helm chart) the state gets reset.

Maybe `/var/lib/elastic-agent-eck/elastic-system/state`?

Reply:
good catch @joshdover! this is now fixed and the state of agents with `daemonset` mode is managed by ECK `/var/lib/elastic-agent/$namespace/$agent_name`. For the agents under `statefulset` or `daemonset` the state is set as an `EmptyDir` to avoid problems like this [one](https://github.com/elastic/cloud-on-k8s/issues/6468)